### PR TITLE
Adding an assertion that the payment method needs to be available.

### LINKED ISSUE #650: [BUG] Missing validation during choosing cart payment method
Example: Lets say we have Method_a and Method_b in the Sylius. But only Method_a is allowed for the channel.

We call [get] /checkout/XXXXXXX/payment and receive Method_a as the only available option. What is correct.

Then we [put] /checkout/XXXXXXX/payment/0 with the body: {"method": "Method_b"}  ( .._b not .._a)

And it works. Method_b becomes payment method for our order what should not happen.

So there should be some validation for this situation.

### PR DESCRIPTION
**Pull request by mamazu**
Fixes #650

### CODE REVIEW

File: src/Handler/Cart/ChoosePaymentMethodHandler.php [hunk 1]
```diff
@@ -24,36 +26,56 @@ final class ChoosePaymentMethodHandler
...
 
+    /** @var PaymentMethodsResolverInterface */
+    private $paymentMethodsResolver;
+
     public function __construct(
         OrderRepositoryInterface $orderRepository,
         PaymentMethodRepositoryInterface $paymentMethodRepository,
-        FactoryInterface $stateMachineFactory
+        FactoryInterface $stateMachineFactory,
+        PaymentMethodsResolverInterface $paymentMethodsResolver
```
Comment (lchrusciel) [inline src/Handler/Cart/ChoosePaymentMethodHandler.php:36]: This argument should be nullable and should trigger deprecation error if not send

Comment (mamazu) [reply src/Handler/Cart/ChoosePaymentMethodHandler.php]: Done. If the property is not set then the validation (aka assertion) will not be triggered as well. So fully backwards compatible.

**Comment by mamazu** [comment]
Then this could be merged.

**Comment by lchrusciel** [comment]
Thank you, @mamazu! :tada:

**Comment by lchrusciel** [comment]
Thanks Max 🏅

**Pull request merged**